### DISCUSSION TITLE
mqtt & rtsp at the same time

### DISCUSSION THREAD

**Discussion by mattlcbesc** [QuantumEntangledAndy/neolink#103] [General]
maybe I am missing something, but is there a way to use MQTT & RTSP functions at the same time?  Seems it's one or the other in my testing so far.  Thanks!!

**Comment by QuantumEntangledAndy** [comment]
Yep one or the other at the moment. Although you can just run both commands at once.

It's mostly a maintenance thing since both of complex I like to work on one then the other to keep things simple.

**Reply by mattlcbesc** [reply]
Alright, i'm going to sound like a noob. How would I accomplish running both at the same time?  I havent had enough caffeine yet.

**Reply by QuantumEntangledAndy** [reply]
Basically just open two terminals and run one neolink command in each terminal

**Comment by wizmo2** [comment]
Have been playing around with a Lumus, and just wanted to chime in.
Firstly, I have to thank Andy and 30 for the effort.  This project is officially brill!

On topic, I get best results running the RTSP and MQTT on different hosts. 

We monitor this camera, along with some older Amcrest devices, mostly through a vpn tunnel.  One of the reasons why I went for the Reolink, is the Amcrests are 'mareish as every camera has different API bugs, and the RTSP support is just plain bad.   I now use [rtsp-simple-proxy](https://github.com/aler9/rtsp-simple-proxy)  which seems to address most of streaming issues, plus significantly reduces the load when accessing  from multiple locations.  

My camera is remote mounted, with suspect wifi connectivity.  I had some initial issues getting the RTSP  functionality stable,  Switching to the pre-configured docker helped loads.   The RPi4 host is working hard when streaming, so I now only use `stream = "subStream"` and slow the frame-rate down to 6-10fps.  

I'm using MQTT connectivity for Home-Assistant.  If you're using HASS-OS, the HA Add-on is basically PNP and IMHO the way to go  (it is essentially another docker thread, but appears to require minimal resources and most traffic is local to the HA host).  With `features = ["led", "ir", "floodlight", "motion", "reboot"]`, all the entities show up automatically.  I added the RTSP stream using the Generic Camera Integration using the proxied stream.

So far I'm impressed with both the Lumus (software, hardware, design) and NEOLINK (of course!).  Thanks again

**Comment by mesistostefre** [comment]
Hopping on this thread ... cause I'd like to understand this a bit better in the context of running neolink as service(s).
- I do have "neolink rtsp ... " running as a windows service successfully
- I also do have "neolink mqtt ..." running as another windows service and I did receive some MQTT messages at the broker

Thing is: the values do not update. I did expect that the "neolink mqtt ..." command would periodically send MQTT messages. Like, when the battery level changes. That does not seem to be the case.

Am I correct that in fact I have to execute "neolink mqtt ..." periodically myself? Like any other minute or so?

Whatever the truth is: maybe a good idea to adjust the doc a bit... ;-)

**Reply by QuantumEntangledAndy** [reply]
Just run neolink mqtt once. It will update when it will. 

Battery is reported:
- on demand with the query message
- when the camera decides to send out events (seems to only do this when crossing low power thresholds)


The only one that is updated regularly is the preview one but that depends on if your camera supports it.

**Reply by mesistostefre** [reply]
Thanks for your comment. I really appreciate.

Ok, means to me: running both as a service should do ...
Strangely enough I only receive the battery level once right after (re)starting "neolink mqtt". Also not when on low power. Bummer...

Thanks anyway and also thanks for your work on this.

**Reply by QuantumEntangledAndy** [reply]
You could probably script something to post the query battery message regularly. Or open up a PR to add it as a feature.

**Reply by mesistostefre** [reply]
Exactly ... I just did a tiny script executed each 10 mins to start and stop the service running the "neolink mqtt ... "
... and another script restarting "neolink rtsp ..." once a day because this helps neolink recover from long time cam downtime.

Both automated restarts are not nice ... it is rather fighting the symptom than the root cause, but all I can do for now and I can live with it.

Thanks for your efforts!

**Reply by QuantumEntangledAndy** [reply]
I didn't mean to script a full restart but a script to post the `/query/battery` message
